packages/clay-time-picker/src/index.tsx

```diff
@@ -238,9 +238,33 @@ const ClayTimePicker: React.FunctionComponent<IProps> = ({
 				break;
 			default:
 				if (regex.test(event.key) && configName !== TimeType.ampm) {
+					const maxFirstDigit =
+						configName === TimeType.hours
+							? use12Hours
+								? 1
+								: 2
```

Comment:
I would say the same for that value 😅

Reply:
Same as above, except for hours. We should probably do programmatically from the config.